solidity/test/InterchainAccountRouter.t.sol

```diff
@@ -124,80 +129,102 @@ contract InterchainAccountRouterTest is Test {
         target = new Callable();
     }
 
-    function testConstructor() public {
-        // The deployed ICA should be owned by the router
-        destinationRouter.getDeployedInterchainAccount(
-            origin,
-            address(this),
-            address(originRouter),
-            address(environment.isms(destination))
-        );
-        assertEq(ica.owner(), address(destinationRouter));
+    function testFuzz_constructor(address _localOwner) public {
+        OwnableMulticall _account = destinationRouter
+            .getDeployedInterchainAccount(
+                origin,
+                _localOwner,
+                address(originRouter),
+                address(environment.isms(destination))
+            );
+        assertEq(_account.owner(), address(destinationRouter));
     }
 
-    function testGetRemoteInterchainAccount() public {
-        assertEq(
-            originRouter.getRemoteInterchainAccount(
-                address(this),
-                address(destinationRouter),
-                address(environment.isms(destination))
-            ),
-            address(ica)
+    function testFuzz_getRemoteInterchainAccount(
+        address _localOwner,
+        address _ism
+    ) public {
+        address _account = originRouter.getRemoteInterchainAccount(
+            address(_localOwner),
+            address(destinationRouter),
+            _ism
         );
         originRouter.enrollRemoteRouterAndIsm(
             destination,
             routerOverride,
-            ismOverride
+            TypeCasts.addressToBytes32(_ism)
```

Comment:
why is the ISM bytes32?

Reply:
no pointed out yourself, this is remote ism